package.json

```diff
@@ -28,11 +28,14 @@
     "changelog": "./bin/plugin/cli.js changelog",
     "since": "./bin/plugin/cli.js since",
     "readme": "./bin/plugin/cli.js readme",
-    "build": "wp-scripts build",
+    "build": "npm-run-all 'build:!(plugin)'",
+    "build:optimization-detective": "wp-scripts build -c plugins/optimization-detective/webpack.config.js",
```

Comment:
It will be run automatically on the pre-hook. See [diff](https://github.com/WordPress/performance/pull/1080/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R38-R39).

Reply:
Oh I see (and sorry for my truncated com

😉

Reply:
> Separately, in Optimization Detective, when first cloning the repo for development, isn't there a check to see if the web-vitals script is present, and if not an admin notice is shown with instructions to build the plugin? Won't that need to be updated?

I think we did it in the `can-load.php` file which no longer exists. We need to add this check in the main plugin file and die with an admin notice if the required assets are not there. Shall I add this as the part of same PR?

Reply:
Oh right. If we think it's important to add, or alternatively what if the asset build happens automatically after npm install?

Reply:
> or alternatively what if the asset build happens automatically after npm install?

I don't see an issue with that. But in case someone passed `--ignore-scripts` flag, that wouldn't help. Given that, I think we can add a safeguard to handle that.

Reply:
Or if a file is missing wouldn't the user learn about it from the error?

Reply:
Yeah, they would get a fatal error on the frontend when attempting to require the asset PHP file. In PHP7 can't this Error be caught? If so, we could then show an error in the console about needing to build.

Reply:
How about this:

```diff
--- a/plugins/optimization-detective/detection.php
+++ b/plugins/optimization-detective/detection.php
@@ -35,8 +35,17 @@ function od_get_detection_script( string $slug, OD_URL_Metrics_Group_Collection
 	 */
 	$detection_time_window = apply_filters( 'od_detection_time_window', 5000 );
 
-	$web_vitals_lib_data = require __DIR__ . '/detection/web-vitals.asset.php';
-	$web_vitals_lib_src  = add_query_arg( 'ver', $web_vitals_lib_data['version'], plugin_dir_url( __FILE__ ) . '/detection/web-vitals.js' );
+	try {
+		$web_vitals_lib_data = require __DIR__ . '/detection/web-vitals.asset.php';
+	} catch ( Error $error ) {
+		return wp_get_inline_script_tag(
+			sprintf(
+				'console.error( %s )',
+				wp_json_encode( '[Optimization Detective] %s' . __( 'Unable to load web-vitals.asset.php. Please make sure you have run `npm install && npm run build:optimization-detective`.', 'optimization-detective' ) )
+			)
+		);
+	}
+	$web_vitals_lib_src = add_query_arg( 'ver', $web_vitals_lib_data['version'], plugin_dir_url( __FILE__ ) . '/detection/web-vitals.js' );
 
 	$current_url = od_get_current_url();
 	$detect_args = array(
```

Reply:
If we are going to add it, I think the plugin screen would be the best place as users will come to know about this as soon as the plugin is activated. But above approach also looks good to me.

Reply:
Added in 4656cbf379061fd96d545cdfb61aa0ab71855af6